main.go

```diff
@@ -484,21 +484,21 @@ func addExtensions(msg *descriptor.DescriptorProto, extensions []*descriptor.Fie
 			continue
 		}
 
-		var foundNode *descriptor.DescriptorProto
-		nestedMsgs := msg.GetNestedType()
-		for i := nodeIdx; i < len(extendeeNodes); i += 1 {
-			foundNode = nil
+		// Keep going deeper, perhaps its a nested msg
+		foundFlag := true
+		for i := nodeIdx + 1; i < len(extendeeNodes); i++ {
+			nestedMsgs := msg.GetNestedType()
+			foundFlag = false
 			for _, nestedMsg := range nestedMsgs {
 				if nestedMsg.GetName() == extendeeNodes[i] {
-					foundNode = nestedMsg
+					msg = nestedMsg
+					foundFlag = true
 					break
 				}
 			}
```

Comment:
Could be wrong but I think the outer loop here is missing a break statement:

```suggestion
			}

			if foundFlag {
				break
			}
```

Reply:
I see you are paying attention!  But no, this is wanted behavior.  We don't want to break the outer-loop because we want to keep going as deep as possible.  It's not because we found a match that it's the last possible one.

Reply:
Right, I missed how `msg` is progressively set to deeper and deeper nested levels. However, the flag behaviour still seems bugged since it gets reset on each iteration of the outer loop. What happens if there's a match at `i == 1` but none at `i == 2`? If I'm reading the code right, `foundFlag` will stay false and the matched node will not be set on line 102.

Reply:
Yep, you are right, that's what happens. Thing is, that's also wanted behavior due to context.  We only want to consider a match if everything matches until the end of the outer-loop

Reply:
Alright! It's unfortunate that you merged this PR already, I would have asked you to add a note documenting that this is the desired behaviour. Without that context, it's hard to understand what the desired behaviour is.

Reply:
One could argue that tested code is the desired behaviour. There are tests that test specifically this behaviour.  But I do get your point, sorry if I merged too soon, I thought everyone was happy with it and honestly really wanted to get past this

Reply:
Sorry if that came across as accusatory, that wasn't the intention! FWIW I disagree with tests serving as documentation, they can definitely be a reference but the implementation should ideally be clear on its own or documented if it can't be. Just wanted to outline that here but it's not a big deal in this case :)